### ISSUE TITLE
Vitabright doesn't work

### ISSUE THREAD

**Issue by prashantkamdar** [devnoname120/vitabright#13]
As the title says - vitabright doesn't work for me.
I've browsed the issues and found a plugin with logs enabled from this thread: https://github.com/devnoname120/vitabright/issues/10
I used it and have attached my log below.
Using the hacks.guide, my "tai" folder in ux0 is deleted.
Only ur0 tai folder exists.
Have also attached the config file that is being used.
vitabright_lut.txt file is present in the tai folder (in ur0) alongwith the config and the vitabright.skprx plugin (hope is the correct place?)

Other specs:
PCH-1103 (OLED)
3.65 h-encore with enso

[vitabright_log.txt](https://github.com/devnoname120/vitabright/files/2847598/vitabright_log.txt)
[config.txt](https://github.com/devnoname120/vitabright/files/2847600/config.txt)
[vitabright_lut.txt](https://github.com/devnoname120/vitabright/files/2847601/vitabright_lut.txt)

**Comment by devnoname120** [comment]
Could you try again with the latest version?
Attached is a debug version.

**Comment by prashantkamdar** [comment]
Found the log at - ur0:/data/vitabright_log.txt

[vitabright_log.txt](https://github.com/devnoname120/vitabright/files/2847951/vitabright_log.txt)

**Comment by devnoname120** [comment]
Interesting, the log suggests that it works properly.
Could you send me the module `os0:/kd/oled.skprx`?

**Comment by prashantkamdar** [comment]
Please rename from txt to skprx the file since github wasn't letting me upload .skprx.

[oled.txt](https://github.com/devnoname120/vitabright/files/2848643/oled.txt)

**Comment by devnoname120** [comment]
There is nothing abnormal in the log, and your module version has the correct offset.
Could you give me the log corresponding to this version?
[vitabright.zip](https://github.com/devnoname120/vitabright/files/2848847/vitabright.zip)

**Comment by prashantkamdar** [comment]
Here you go:
[vitabright_log.txt](https://github.com/devnoname120/vitabright/files/2849065/vitabright_log.txt)

**Comment by devnoname120** [comment]
I wonder if this might be a taiHEN issue because as far as the log is concerned, everything looks alright.

**Comment by prashantkamdar** [comment]
I just found out about IMCUnlock after I googled it after you wrote it here. Didn't even know about this earlier.
So no, I am not using IMCUnlock.

I've requested a friend for his hacked vita so that I can test your plugins there.
Will run the same diagnostics and see what happens.

Will report back in 5-7 hours as of writing this.

**Comment by prashantkamdar** [comment]
I've finished trying several things and none of them work.
For all of the below test cases, I've used the latest test build provided by you in this thread.

1) In one of your other issue threads, you recommended using usb transfer if wifi transfer has been used. Did that, didn't work.

2) I hard reset my console (power+ps+R) and formatted my memory card. Installed h-encore with help from final-he and setup vitabright in ux0 tai with the lut file. Didn't work.

3) After #2, I setup vitabright in ur0 as recommended in vita hacks guide (without deleting ux0 tai). Didn't work.

4) Same as #3, but with ux0 deleted. Didn't work.

5) Installed enso (because what's few more test cases?). Didn't work.

6) Used the latest debug build in this thread on my friend's vita on 3.68. WORKED!

At this point, the only different things are that my friend's vita is on 3.68 and mine on 3.65.
Below I've attached both the logs.

Am also attaching the screenshot with both vitas. Top is the working one with already turned down brightness; lower is mine :(

I really appreciate your help, can you do anything further?
Any other build with more number of logs, maybe?

[notworkingvitabright_log.txt](https://github.com/devnoname120/vitabright/files/2852580/notworkingvitabright_log.txt)
[workingvitabright_log.txt](https://github.com/devnoname120/vitabright/files/2852581/workingvitabright_log.txt)

![photo 12-02-19 12 09 01 am](https://user-images.githubusercontent.com/2802365/52586095-9d27a200-2e5c-11e9-97c8-647975175909.jpg)

**Comment by devnoname120** [comment]
I'm not sure what happens, could you try this?
https://www.reddit.com/r/vitahacks/comments/aohjz2/vitabright_10_now_supports_lcd_screens_ps_vita/egc8221/

**Comment by prashantkamdar** [comment]
I just did that.
Still doesn't work.

**Comment by prashantkamdar** [comment]
I found somewhere on reddit that the first version works.
So I tried 0.01 version from your releases.

I had the lowest setting of brightness on and then I rebooted.

After rebooting the console and going to the brightness settings menu, the brightness jumps to slightly low than before.

So ya, your note of 

> same brightness as when the screen dims after some inactivity
seems to be working on my vita with 0.01 release.

**Comment by prashantkamdar** [comment]
Do you think logging `lut_inject` from https://github.com/devnoname120/vitabright/blob/73a8af51b2a9fbca6b1914ef3808f2b0a4ca0902/oled/hooks.c#L88 will help?

**Comment by devnoname120** [comment]
Indeed, this is fixed now.

Can you give me the logs that this version produces?
[vitabright.zip](https://github.com/devnoname120/vitabright/files/2930993/vitabright.zip)

**Comment by prashantkamdar** [comment]
`vitabright started...
[OLED] getmodninfo: 0x00000000
[OLED] modid: 0x000100F7
[OLED] Size ok, hooking...
[OLED] OS version: 0x03650000
, table offset: 0x00001E00, ksceOledGetBrightness_addr: 0x000012BD, ksceOledSetBrightness_addr: 0x00000F45
[OLED] injectdata: 0x00011E81
[LCD] getmodninfo: 0x90010002
[LCD] modid: 0x00011651
`

**Comment by prashantkamdar** [comment]
different log!
[vitabright_log.txt](https://github.com/devnoname120/vitabright/files/2931971/vitabright_log.txt)

**Comment by araujojr82** [comment]
Hello guys, I'm facing the same problem here.

My PSVita is version PCH-1001. I've tried just about everything on this topic and more. I was using 3.68 with h-encore but after no success, I thought that maybe enso would help, so I downgraded to 3.65 and using h-encore with enso. That didn't work either.

Also, I've tried with AND without SD2VITA.

Here are some of my files.
os0:/kd/oled.skprx:
[oled.txt](https://github.com/devnoname120/vitabright/files/2932064/oled.txt)

Log from 1.0 version:
[vitabright_log_1_0.txt.txt](https://github.com/devnoname120/vitabright/files/2932066/vitabright_log_1_0.txt.txt)

Log from today's version:
[vitabright_log.txt](https://github.com/devnoname120/vitabright/files/2932065/vitabright_log.txt)

I hope this helps.

**Comment by SDM855** [comment]
Facing the same problem

Mine is PCH-1000 Japanese version(3.60 enso). This plugin works on my PCH-2000(3.60 enso) without any problem.

And I cannot find "data" folder under ur0, I cannot find it on PCH-2000 either(although it works).

The attachment is oled.skprx, hope it helps
[https://github.com/SDM855/PSV/blob/master/oled.skprx](url)

**Comment by araujojr82** [comment]
> 
> 
> Facing the same problem
> 
> Mine is PCH-1000 Japanese version(3.60 enso). This plugin works on my PCH-2000(3.60 enso) without any problem.
> 
> And I cannot find "data" folder under ur0, I cannot find it on PCH-2000 either(although it works).
> 
> The attachment is oled.skprx, hope it helps
> [oled.txt](https://github.com/devnoname120/vitabright/files/2933956/oled.txt)

Just create the "data" folder using VitaShell in ur0 and use one of those versions with activated log and it shall produce some log files on the folder.

**Comment by SDM855** [comment]
Thanks to araujojr82's help
I got the vitabright_log.txt (for today's version)
hope it helps
[vitabright_log.txt](https://github.com/devnoname120/vitabright/files/2934517/vitabright_log.txt)

**Comment by devnoname120** [comment]
All your logs indicate that the table injection works properly...
My bet is some sort of issue in taiHEN or some kind of race condition.

**Comment by SDM855** [comment]
> All your logs indicate that the table injection works properly...
> My bet is some sort of issue in taiHEN or some kind of race condition.

Tried on formatted new system, still no luck...
Just curious, as I see in above attachments, the working modid is 0x000100F5, ours are 0x000100F7. Is that means Sony use different OLED panel?

**Comment by devnoname120** [comment]
I don't think so, this is a UID so it should depend on the module load order.
I've confirmed that some people have `0x000100F5` and others `0x000100F7` and it works for both of them.

**Comment by maskmark** [comment]
right.it works on one of my 1000，the screen goes orange by default.
on my other 1000，it just do nothing.
they are both on 3.65 encore enso.I even tryed use a ux0 config.txt with only this plugin to test both one. and yes, still worked for one and do nothing on the other.

**Comment by darkkomdld** [comment]
i just have the same problem here: One of my consoles dont run it but!!!! being enso 3.60 both in that one i cant find repatch for 3.65  or make it works. I dont knwo if it relate. Thx we dont just need repach only nowadays. They both model are 1004. I just find one difference the one is failing its serial code 2A, the one is working is serial model 2D. So they are same model but the one its no working its the very first generation.

**Comment by darkkomdld** [comment]
i need help....

**Comment by junhawng** [comment]
[vitabright_log.txt](https://github.com/devnoname120/vitabright/files/3067833/vitabright_log.txt)
I get a parsing error with my vita 1000. What could be the problem?  Thank you.

**Comment by devnoname120** [comment]
@junhawng Your `vitabright_lut.txt` file is incorrect, make sure that you follow the formatting and that you use Unix line endings (`\n`), and not Windows line endings.

**Comment by junhawng** [comment]
> @junhawng Your `vitabright_lut.txt` file is incorrect, make sure that you follow the formatting and that you use Unix line endings (`\n`), and not Windows line endings.

I was able to get it working but I'm not able to notice an actual difference with the display brightness. How can I be certain it's working?

**Comment by Padankadank** [comment]
Just throwing my troubles in here too, PCH-1101, 3g oled vita 1000. On 3.65 h-encore.

I used the debug version, it says it runs correctly but I dont see any noticeable difference when sliding the brightness slider all the way down

**Comment by hamodehsabie** [comment]
The issue is the screen itself. I have 2 vitas one works with v1 and the other one on works with the first version (semi working one) 
I swapped motherboards and the sides boards and both devices boards works with the plugin with 1 screen the other screen panel doesnt work with vitabright 
And i even tried 3.60 , 3.65 , 3.68 firmwares on both devices with correct setup of the plugin using auto plugin. 
Please try to find a solution for it cheers!

**Comment by devnoname120** [comment]
@hamodehsabie If this is not too much of a bother, could you try just swapping the two screens? This would help confirm beyond doubts that the screen is indeed the root cause.

Also, please give me any serial number of information that you can get about the screens (maybe they are different models and one of them doesn't support custom gamma tables?).

**Comment by hamodehsabie** [comment]
This is exactly what i was trying to say i swapped the 2 screens and the boards and only one screen was working with vita bright 
Both vitas are OLED 1101 model

**Comment by devnoname120** [comment]
Interesting. Anything about the serials or any numbers appearing on the physical screens?

**Comment by hamodehsabie** [comment]
Sorry for my long replies, i will reply faster now
And what do you mean on the physical screens ? 
Because i swapped the whole front panels u cant see the back of the screen its covered from the back on the screen cable is visible 
But both devices are the same even the same bundle with walking dead game card gift 
Its really strange because i thought its the mother board issue but when i swapped both boards worked with vita bright on ghat screen and both boards did not work on the 2nd screen

**Comment by devnoname120** [comment]
It's possible that some screens don't accept new gamma tables. The semi-working version wasn't replacing the gamma table, it was only forcing the "dimmed brightness after inactivity" level to be enabled even when there is activity.

**Comment by hamodehsabie** [comment]
Update: i confirm the Screens version of the 2 OLED model 1101 are different the one with the model number ( OLED B313 ) is the one that works with all versions of vita bright 
The other screen version is (OLED B 106 ) and that one only works with the semi working version of Vitabright 
I open the devices again and i notice the models on the back of the screen cable.

**Comment by devnoname120** [comment]
Hm that's interesting. I would love to have confirmation from other users to make sure that OLED B106 never works and OLED B313 always works.

Could you take a picture of these two references? This would allow other users to find their reference more easily.

**Comment by WBSpitfire** [comment]
I took apart my Vita looking for the OLED model, but was unable to find it. Do you have to "bake" the front panel to separate the adhesive to see the model? Or am I just blind?

**Comment by UniLastation** [comment]
> Hm that's interesting. I would love to have confirmation from other users to make sure that OLED B106 never works and OLED B313 always works.
> 
> Could you take a picture of these two references? This would allow other users to find their reference more easily.

I can confirm that I have an OLED model B106-23704 for a PCH-1001, and an OLED model AB29-03039 for a PCH-1000. Both of which do not work for any but the 0.01 release of VitaBright.

**Comment by Padankadank** [comment]
Just wanted to confirm that only v0.1 works on my vita, im having the same problem as everybody else here

**Comment by devnoname120** [comment]
@Padankadank Could you check the OLED model of your vita? You can see above how others were able to find it.

**Comment by hamodehsabie** [comment]
WBSpotfire: yes this is where i located it but only on the red circled cable because this cable is visible after removing the boards but the where is the green circled part you have to remove the screen panel from the front to look at it which mean removing the the and such things  

Devnoname: i cant confirm now because the working one is with its owner now 
And i dknt want to keep opening the devices because i have to remove all boards again and again and i dont want to break anything by mistake im really sorry but i guess there is a-lot of versions  according to the numbers people mentioned  above maybe? 
Its just some screens doesnt support it i guess...

**Comment by adrianbass360** [comment]
Dang, I don't get how to change the gamma ratio on this application. Can someone do one where the vita retains it's originals colors, just with both higher and lower brightness levels?

**Comment by olokos** [comment]
I have the same issue. I already tried workarounds that worked for others including [disabling 3G in registry](https://www.reddit.com/r/vitahacks/comments/96p9zk/vita_bright_works_on_368_cfw/e434anj/) and going into flight mode. 3.65 Enso

Log for vitabright 1.0-debug

[vitabright_log.txt](https://github.com/devnoname120/vitabright/files/3416863/vitabright_log.txt)
Log for the second vitabright plugin from this issue

[vitabright_log.txt](https://github.com/devnoname120/vitabright/files/3416877/vitabright_log.txt)
If I get this right I should be having an orange hue like on really low brightness levels, but I can't really see any difference in either lowest or highest brightness. Until I saw this topic I thought it's just not that obvious that it's working, but now I'm positive it isn't. Brightness levels and colors are the same with or without the plugin.

**Comment by devnoname120** [comment]
@olokos 
1) Could you try again with this version? https://github.com/devnoname120/vitabright/issues/13#issuecomment-469671030

2) Could you give me all the serial numbers that appear on the physical OLED screen? You can scroll a bit up to find out where other users found them (this will require you to open your Vita).

**Comment by olokos** [comment]
@devnoname120 
1. Certainly 
[vitabright_log.txt](https://github.com/devnoname120/vitabright/files/3424361/vitabright_log.txt)
I might have mixed them up, even thought I checked it multiple times, now it's surely spot on, just pulled it off device.

2. I can do it, but I can't tell exactly when, hopefully by the next week.

**Comment by devnoname120** [comment]
Alright I don't see anything wrong in the log. Looking forward to your OLED numbers.

**Comment by CelesteBlue-dev** [comment]
To better identify the PSVita which could have such a weird screen,
I ask each of you to open modoru app by TheFloW and to report the minimal FW version it tells you. For exemple my main Psvita could go down to 1.61. My other PSVita 1.06.

**Comment by olokos** [comment]
@CelesteBlue-dev took me a while to disable all plugins, but hey, still miles easier than disassembling the thing!

Sadly after doing so it updated my database leaving me with nothing but default apps. I keep near app replaced with vitashell as fallback in case of this kind of situation. I'm away on holidays until around Monday so this is typed on mobile. 

Is there some other good way of checking minimum firmware version? Won't let me check it because of plugins, won't let me start it as it's gone once only henkaku and adrenaline plugins are enabled. I can try installing modoru again after all plugins are gone, but it will be much safer when I'm actually back home.

**Comment by CelesteBlue-dev** [comment]
@olokos I don't understand. Launch modoru with L and R being pressed. Read the version then close modoru. Don't attemt downgrade it' not useful.

**Comment by WBSpitfire** [comment]
Minimal firmware version for my Vita 1001 is 1.06.

**Comment by olokos** [comment]
Same here, minimal firmware version 1.06. I have replacement battery on the way to me now, I'll check the model under the screen once it arrives.

**Comment by JosePepowner** [comment]
Same issues as above. I just pulled my Vita out my tech bag after over 6 years and realized this screen is super bright. And yes I installed everything correctly. All other plugins work perfectly. Although I just did all the hacking yesterday I know it's all good. Been rooting phones since jellybean and that can get way more confusing. 

My Vita is an American day 1 release. If that helps at all. I'm not trying to push Dev on this but it would be cool if my screen was supported. This thing is fucking bright on low. And honestly might be sunglasses-inside-at-night-douchebag-man for a while.

**Comment by LuanTeles** [comment]
I got the same problem but with a slim one, the minimum works, but the maximum didnt increase a bit

**Comment by devnoname120** [comment]
That's because it's not possible to increase it further on the slim.

**Comment by prashantkamdar** [comment]
@devnoname120 
I see this issue/bug still open since Feb.
Anything I can do to help you resolve? 
Replicate the issue?

**Comment by devnoname120** [comment]
Unfortunately no. I don't know where the problem comes from and I haven't been able to get my hand on a unit in order to debug it.

**Comment by RanCode** [comment]
@devnoname120 My device has same problem, I tried 0.01, 0.5, 0.9 and 1.0 version, but only 0.01version worked, you know, 0.01version only dim a bit, could you please develop new vesion based on 0.01version, thanks a lot!

**Comment by Feardarks** [comment]
still v1 doesnt works.

**Comment by olokos** [comment]
@devnoname120 
I am willing to do some debugging, if I don't have to tear apart the device too much! :)
It seems I would have to unglue the screen in order to reach the sticker on the display, as it's hidden beneath this plastic plate :(
My vita is in a really good condition as I was hunting for a good one for quite a bit, I am afraid it won't be after I'm done reaching that serial number. :(

**Comment by CelesteBlue-dev** [comment]
I have found 2 new methods to better know the PSVita manufacturing date:
- on the motherboard, if you take a good quality photo, you can see the year and week of manufacturing. Example: if it is written 12 50 it means 50rd week of year 2012.
- In the eMMC first partitions, specifically idstorage partition, there is a timestamp. I will make a dumper someday.

**Comment by olokos** [comment]
Where exactly can I find that number?

I can take a photo if I won't have to rip the screen off the plastic :D

Can I somehow get the timestamp without the dumper?

**Comment by SeongGino** [comment]
Just going to add to this pool here, but slightly different from the rest.

Up until now, I was using my launch model WiFi Vita as-was. However, the passage of time wasn't kind to the display and was scratched to hell. Conveniently enough, Vitabright sort of helped with seeing the screen through the softening caused by marked glass, but I wasn't going to live with it forever.

Today (and by today, I mean within the last hour of me posting this), I managed to get a replacement 'OEM' (that's what the seller claims, anyways) front housing with panel + digitizer. One disassembly later, and I got the thing all together with the new display. Unfortunately, it looks like despite having Vitabright enabled, it's reverted back to stock display brightness range. So I guess I can count myself as one of the unlucky few who used to have a compatible screen and changed to a non-compat one, all on the same base system.

Before asking, I couldn't see what 'model' it is, and I have no interest in baking my now-mint-condition front just to find out. As cheap as the housings are, times are a little too tough to go about spending more on unnecessary replacements and lamination.

**Comment by falkenhawk** [comment]
v1.0 does not work on neither of my two `PCH-1004` vitas :((( Can I help with debugging?

**Comment by falkenhawk** [comment]
I reinstalled with the version from https://github.com/devnoname120/vitabright/issues/13#issuecomment-469671030 and... it started working on **one** of my vitas! Previously I installed the plugin on both with Autoplugin II, so maybe something got borked there.

The one which works, is on 3.60, and is possibly a newer hardware than the other non-working one, which is currently on 3.65 (I doubt that the fw version makes any difference here), because I bought that one on day 1 in EU, and the one working with vitabright somewhen later.

Logs from the working one:
```
vitabright started...
getmodninfo: 0x00000000
modid: 0x000100F7
Size ok, hooking...
injectdata: 0x00011BC9
vitabright started...
getmodninfo: 0x00000000
modid: 0x000100F7
Size ok, hooking...
injectdata: 0x00011BD3
vitabright started...
getmodninfo: 0x00000000
modid: 0x000100F7
Size ok, hooking...
injectdata: 0x00011BCD
vitabright started...
getmodninfo: 0x00000000
modid: 0x000100F7
Size ok, hooking...
injectdata: 0x00011BCD
vitabright started...
getmodninfo: 0x00000000
modid: 0x000100F7
Size ok, hooking...
injectdata: 0x00011BCD
vitabright started...
getmodninfo: 0x00000000
modid: 0x000100F7
Size ok, hooking...
injectdata: 0x00011BCD
vitabright started...
getmodninfo: 0x00000000
modid: 0x000100F7
Size ok, hooking...
injectdata: 0x00011BD5
vitabright started...
getmodninfo: 0x00000000
modid: 0x000100F7
Size ok, hooking...
injectdata: 0x00011BD5
vitabright started...
getmodninfo: 0x00000000
modid: 0x000100F7
Size ok, hooking...
injectdata: 0x00011BC9
vitabright started...
getmodninfo: 0x00000000
modid: 0x000100F7
Size ok, hooking...
injectdata: 0x00011BCB
vitabright started...
[OLED] getmodninfo: 0x00000000
[OLED] modid: 0x000100F7
[OLED] Size ok, hooking...
[OLED] OS version: 0x03600000
, table offset: 0x00001E00, ksceOledGetBrightness_addr: 0x000012BD, ksceOledSetBrightness_addr: 0x00000F45
[OLED] injectdata: 0x00011FA5
[LCD] getmodninfo: 0x90010002
[LCD] modid: 0x000115C5
```

Logs from the non-working one:
```
vitabright started...
[OLED] brightness table parsing failure
[LCD] getmodninfo: 0x90010002
[LCD] modid: 0x000117C1
vitabright started...
[OLED] getmodninfo: 0x00000000
[OLED] modid: 0x000100F7
[OLED] Size ok, hooking...
[OLED] OS version: 0x03650000
, table offset: 0x00001E00, ksceOledGetBrightness_addr: 0x000012BD, ksceOledSetBrightness_addr: 0x00000F45
[OLED] injectdata: 0x0002150D
[LCD] getmodninfo: 0x90010002
[LCD] modid: 0x000117B7
vitabright started...
[OLED] getmodninfo: 0x00000000
[OLED] modid: 0x000100F7
[OLED] Size ok, hooking...
[OLED] OS version: 0x03650000
, table offset: 0x00001E00, ksceOledGetBrightness_addr: 0x000012BD, ksceOledSetBrightness_addr: 0x00000F45
[OLED] injectdata: 0x000214F3
[LCD] getmodninfo: 0x90010002
[LCD] modid: 0x000117AD
```

**Comment by SeongGino** [comment]
Unfortunately, that isn't the case on mine.
```vitabright started...
[OLED] getmodninfo: 0x00000000
[OLED] modid: 0x000100FD
[OLED] Size ok, hooking...
[OLED] OS version: 0x03600000
, table offset: 0x00001E00, ksceOledGetBrightness_addr: 0x000012BD, ksceOledSetBrightness_addr: 0x00000F45
[OLED] injectdata: 0x00011EE3
[LCD] getmodninfo: 0x90010002
[LCD] modid: 0x000116AF```

**Comment by falkenhawk** [comment]
FWIW, I downgraded the non-working one from 3.65 to 3.60 to check if it will make any difference. It didn't, it still does not work. I can post logs later if it gives any value.

**Comment by prashantkamdar** [comment]
Guys,

I am the original bug reporter.
If you read through the issue in detail, it has been established that this is not a software problem (yet).
The differences lie in the screen type itself.
Due to how some screens operate, the vitabright plugin may or may not work.
I have 3 100x vitas and vitabright works on only 1 of them.

**Comment by SeongGino** [comment]
> 
> 
> Guys,
> 
> I am the original bug reporter.
> If you read through the issue in detail, it has been established that this is not a software problem (yet).
> The differences lie in the screen type itself.
> Due to how some screens operate, the vitabright plugin may or may not work.
> I have 3 100x vitas and vitabright works on only 1 of them.
> 
> Also, the 0.1v of the plugin termed as "working" is just
> 
> > forcing the "dimmed brightness after inactivity" level to be enabled even when there is activity
> 
> It isn't injecting the new lut table so in essence, it's not really a "working" version.

But that's the entire point of the issue, though, isn't it?

Especially when the only workaround, as it stands, is to buy _more_ pricy front housing+displays, perform a full disassembly and reassembly, and then _hope_ that the display is not only still OLED (and a good quality one), but the _right kind_ of OLED. . . devaluing your own issue seems a bit counterintuitive, no?

I'm still going to wait and see if something, however long it takes, comes up so we can get proper lut table injection on more than just the lucky OLED panels.

**Comment by prashantkamdar** [comment]
hey, I wasn't meaning to devalue the issue.
all that you just said, I share the same view.
the only reason I posted it because I felt some recent comments didn't understand the entire history of the issue and how it has transpired to a more hardware issue than a software one.
I, too, am waiting for this issue to get resolved, but I believe the solution isn't going to be a proper lut table, but some core changes in the plugin.

**Comment by Warp1n** [comment]
Yeah it doesn't work on my vita I tried Reddit and discord henkaku but no success.

**Comment by seewood** [comment]
Hello, maybe i found the solution. Can someone test it for me?
I have reset the Battery with the Battery_Fixer from SKGleba and now it works for me... after the battery fix i changed the skprx file, and no longer have the old one. But it seems to be fixed. It is a Vita 1000 without 3G. 

Please do the following and report back:
- Install the Battery Fix: https://github.com/SKGleba/PSP2-batteryFixer/releases
- Start it and dont do anything until the Vita shuts off. Wait 6 seconds and start the Vita. 
- copy the 1.0 release to you config folder (e.g. ur0:tai / ux0:tai

Update me if it works. Maybe it is a Battery related Problem.

**Comment by hamodehsabie** [comment]
I will test and reply back 👍

Sabie.

**Comment by falkenhawk** [comment]
no difference here, @seewood 's trick did not help

**Comment by olokos** [comment]
I also replaced  Batteryfix by @SKGleba on my original old battery and replaced it with another one later, but it made no difference.

I have the Fat one with 3G.

I believe the issue has to be that there are 2 different panels made for PSVita OLED 3G, thus why it works for some of people and doesn't for others.

I saw this before with Sony Xperia Z1, touchscreen wouldn't work for me on AOSP ROMs (open source), constantly had ghost touches or not registering at all but did on Stock ROMs.
Months later it turned out that there was indeed a second digitizer used in that smartphone, which was later introduced to the code, fixing touchscreen for good and I believe this is what is going on here.

Possibly Sony made 2 revisions of the OLED display for Vita aswell?

Or possibly one version is the original one and another used as a replacement, but also used by Sony in their warranty repairs?

**Comment by zekepliskin** [comment]
I too have the issue where newer versions of VitaBright don't work.

I'm using it on a PCH-1003 model (non-3G Phat) running ensō 3.65 from coldboot.

I tried 1.0, 0.9, and the debug/comment #13 versions posted here, no joy.

The only one which does work is the oldest, 0.01.  This is fine but it would be nice if it didn't drop the brightness to minimum every time the Vita is put on sleep.  Still, I downloaded it so I could run NFS Most Wanted and a few others with PSVshell v1.1 controlled overclocking and full brightness (aka **Battery Be Damned!** mode) and it does that so partial success is better than none!  :-D

**Comment by EeugeneGG** [comment]
Hello, on mine Ps vita PCH-1104 (3.65), doesn't work vita bright(((

**Comment by felixsrg** [comment]
Hello, PS Vita Fat PCH-1001 (On Enso Henkaku 3.60) also not working. But I kind of found a workaround for those of us who use the 0.01 version to set the dim brightness.

What I did was to use the StayBright plugin alongside this one and now my Vita stays on its dim brightness all the time after setting it on the quick menu.

It might not be the best solution but at least it's something for the time being.

**Comment by g1ya** [comment]
I have the 3g 1000 model and i can't get it work , only the first version works bit when i put it into sleep mode i have to reset the brightness to the lowest to get work again

**Comment by legomeaker101** [comment]
hey guys so i wanted to add something that could be of some use: i used to have the plugin working perfectly until my screen broke and i got a replacement. i got it from ali express and after installing it the plugin stopped working i tried everything but nothing worked except the 0.1 version.

hope this helps bring support to these oled panels.

**Comment by devnoname120** [comment]
Anyone in Paris willing to lend me a PS Vita for which vitabright doesn't work?
Alternatively, could someone mail one to me?

**Comment by felixsrg** [comment]
I sadly live in South America and can't really mail one of the PS Vitas that I have and on which VitaBright doesn't work, but if it can be helpful, I can help you try and debug on it.

**Comment by devnoname120** [comment]
Rise and shine, new debugging logs for vitabright.
[vitabright.zip](https://github.com/devnoname120/vitabright/files/5513747/vitabright.zip)

@felixsrg @legomeaker101 @zekepliskin @SeongGino @falkenhawk Can you test this version?

Then:
- Reply and drag & drop here the file `ur0:/data/vitabright_log.txt`.
- Say if vitabright works for you or not.

Others you're welcome to do that as well.

**Comment by devnoname120** [comment]
I start. I have the following OLED model and it works with vitabright:

```
[OLED] supplier_id: 0x0000, supplier_elective_data: 0x0806
```

**Comment by felixsrg** [comment]
> Rise and shine, new debugging logs for vitabright.
> [vitabright.zip](https://github.com/devnoname120/vitabright/files/5513747/vitabright.zip)
> 
> @felixsrg @legomeaker101 @zekepliskin @SeongGino @falkenhawk Can you test this version?
> 
> Then:
> 
> * Reply and drag & drop here the file `ur0:/data/vitabright_log.txt`.
> * Say if vitabright works for you or not.
> 
> Others you're welcome to do that as well.

Thank you for your hard work, I've just tried this one on my OLED PS Vita, here's the requested info:

- [vitabright_log.txt](https://github.com/devnoname120/vitabright/files/5513982/vitabright_log.txt)
- VitaBright is still not working, sadly.

My Vita is the following model:

`[OLED] supplier_id: 0x0000, supplier_elective_data: 0x0805`

**Comment by zekepliskin** [comment]
> Rise and shine, new debugging logs for vitabright.
> [vitabright.zip](https://github.com/devnoname120/vitabright/files/5513747/vitabright.zip)
> 
> @felixsrg @legomeaker101 @zekepliskin @SeongGino @falkenhawk Can you test this version?
> 
> Then:
> 
> * Reply and drag & drop here the file `ur0:/data/vitabright_log.txt`.
> * Say if vitabright works for you or not.
> 
> Others you're welcome to do that as well.

Nice of you to say hello and grace us with a new version, thank you.  My version of the log is even less successful than the other two people who checked in, on a model PCH-1003.

We have :-

[code]vitabright started...
[OLED] brightness table parsing failure
[LCD] getmodninfo: 0x90010002
[LCD] modid: 0x000213A9
vitabright started...
[OLED] brightness table parsing failure
[LCD] getmodninfo: 0x90010002
[LCD] modid: 0x0002114B
vitabright started...
[OLED] brightness table parsing failure
[LCD] getmodninfo: 0x90010002
[LCD] modid: 0x00021483
vitabright started...
[OLED] brightness table parsing failure
[LCD] getmodninfo: 0x90010002
[LCD] modid: 0x000214A3
[/code]

So we're not even getting a valid [OLED] supplier ID.  Wonder why.

**Comment by legomeaker101** [comment]
> Rise and shine, new debugging logs for vitabright.
> [vitabright.zip](https://github.com/devnoname120/vitabright/files/5513747/vitabright.zip)
> 
> @felixsrg @legomeaker101 @zekepliskin @SeongGino @falkenhawk Can you test this version?
> 
> Then:
> 
> * Reply and drag & drop here the file `ur0:/data/vitabright_log.txt`.
> * Say if vitabright works for you or not.
> 
> Others you're welcome to do that as well.

I am sadly on the same boat as felix so I cannot lend you my vita (would if I could) so here is what I got:

[vitabright_log.txt](https://github.com/devnoname120/vitabright/files/5518592/vitabright_log.txt)

it is not working.

**Comment by devnoname120** [comment]
@zekepliskin Can you put this file in `ur0:/tai/`? This file is either missing or wrong.
[vitabright_lut.txt](https://github.com/devnoname120/vitabright/files/5519846/vitabright_lut.txt)

**Comment by devnoname120** [comment]
@legomeaker101 @zekepliskin Can you guys send me a message on Discord? I'm on the [HENkaku server](https://discord.gg/8aHMCKT) with the nickname @devnoname120.

**Comment by falkenhawk** [comment]
@devnoname120 logs from non-working one: [vitabright_log.txt](https://github.com/devnoname120/vitabright/files/5520188/vitabright_log.txt)

```
vitabright started...
[OLED] getmodninfo: 0x00000000
[OLED] modid: 0x000100F7
[OLED] Size ok, hooking...
[OLED] OS version: 0x03600000
, table offset: 0x00001E00, ksceOledGetBrightness_addr: 0x000012BD, ksceOledSetBrightness_addr: 0x00000F45 ksceOledGetDDB_addr: 0x0000054D
[OLED] injectdata: 0x00011EBD
[OLED] supplier_id: 0x0000, supplier_elective_data: 0x0805
[LCD] getmodninfo: 0x90010002
[LCD] modid: 0x0001168B
```
and now logs from the working one: [vitabright_log.txt](https://github.com/devnoname120/vitabright/files/5520217/vitabright_log.txt)
```
vitabright started...
[OLED] getmodninfo: 0x00000000
[OLED] modid: 0x000100F7
[OLED] Size ok, hooking...
[OLED] OS version: 0x03600000
, table offset: 0x00001E00, ksceOledGetBrightness_addr: 0x000012BD, ksceOledSetBrightness_addr: 0x00000F45 ksceOledGetDDB_addr: 0x0000054D
[OLED] injectdata: 0x00011DF9
[OLED] supplier_id: 0x0000, supplier_elective_data: 0x0806
[LCD] getmodninfo: 0x90010002
[LCD] modid: 0x000115C5
```

**Comment by devnoname120** [comment]
@falkenhawk that's helpful, can you send me a message on Discord? I'm on the HENkaku server with the nickname @devnoname120.

**Comment by olokos** [comment]
```
vitabright started...
[OLED] getmodninfo: 0x00000000
[OLED] modid: 0x000100F7
[OLED] Size ok, hooking...
[OLED] OS version: 0x03650000
, table offset: 0x00001E00, ksceOledGetBrightness_addr: 0x000012BD, ksceOledSetBrightness_addr: 0x00000F45
[OLED] injectdata: 0x00021555
[LCD] getmodninfo: 0x90010002
[LCD] modid: 0x000118ED
vitabright started...
[OLED] getmodninfo: 0x00000000
[OLED] modid: 0x000100F7
[OLED] Size ok, hooking...
[OLED] OS version: 0x03650000
, table offset: 0x00001E00, ksceOledGetBrightness_addr: 0x000012BD, ksceOledSetBrightness_addr: 0x00000F45
[OLED] injectdata: 0x00021553
[LCD] getmodninfo: 0x90010002
[LCD] modid: 0x000118E9
vitabright started...
[OLED] getmodninfo: 0x00000000
[OLED] modid: 0x000100F7
[OLED] Size ok, hooking...
[OLED] OS version: 0x03650000
, table offset: 0x00001E00, ksceOledGetBrightness_addr: 0x000012BD, ksceOledSetBrightness_addr: 0x00000F45
[OLED] injectdata: 0x00021579
[LCD] getmodninfo: 0x90010002
[LCD] modid: 0x0001190D
```

I'm not sure if it works, but it's not really dim when on minimum brightness.
OLED 3G model

**Comment by Nach18** [comment]
Same to me, I dont know what to do, Why does this happen?

**Comment by pan-theo** [comment]
**Hello everyone!** 

I'm excited to announce that (I think) I have found the solution! 
To me it seems that this is the solution for everyone, because I have a 100X model, did the exact correct steps, and it didn't work, like everyone here, but I have yet to see the results of my supposed fix in other people's cases so I'm not certain, so I'm sorry if this doesn't fix it for you. But I'm positive it's a fix for everyone. 

SO, simply, just like NoNpDrm, the plugin didn't work in ur0, so I transferred it to ux0, and edited the config.txt file appropriately, and it **WORKED!**

Thanks for this plugin, really, now my eyes can be comfortable, and rest, and I can use the Vita healthily.

**Comment by olokos** [comment]
I just tried your idea @pan-theo but it doesn't seem to change anything.

Either the difference it too small or it's still not working with ux0/tai/vitabright.skprx either for me

The vitabright-lut-editor displays brigthness levels from 15(dimmest I can do) to 0(brightest I can do), are those the stock brightness levels or those achieved by vitabright?

**Comment by pan-theo** [comment]
@olokos yo, sorry to hear that. Your issue seems to be exactly what I recently thought is a mistake that should be addressed. 

For some reason, in the GitHub page of the plugin, the developer has accidentally put a "/" where there shouldn't be one, in the instructions of what to write under *KERNEL in config.txt, in "```
ur0:/tai/vitabright.skprx
```" there should not be a "/" before "tai". And also, it only works in ux0 not ur0, at least for some people including us. 

The line should be: ```
ux0:tai/vitabright.skprx
``` and make sure the plugin is actually placed in ux0/tai/

**Comment by pan-theo** [comment]
Code: `line_test`

**Comment by pan-theo** [comment]
@olokos yo, sorry to hear that. Your issue seems to be exactly what I recently thought is a mistake that should be addressed. 

For some reason, in the GitHub page of the plugin, the developer has accidentally put a "/" where there shouldn't be one, in the instructions of what to write under *KERNEL in config.txt, in "```
ur0:/tai/vitabright.skprx
```" there should not be a "/" before "tai". And also, it only works in ux0 not ur0, at least for some people including us. 

The line should be: `ux0:tai/vitabright.skprx` and make sure the plugin is actually placed in ux0/tai/

**Comment by pan-theo** [comment]
@olokos yo, sorry to hear that. Your issue seems to be exactly what I recently thought is a mistake that should be addressed. 

For some reason, in the GitHub page of the plugin, the developer has accidentally put a "/" where there shouldn't be one, in the instructions of what to write under *KERNEL in config.txt, in "ur0:/tai/vitabright.skprx" there should not be a "/" before "tai". And also, it only works in ux0 not ur0, at least for some people including us. 

The line should be: `ux0:tai/vitabright.skprx` and make sure the plugin is actually placed in ux0/tai/

**Comment by olokos** [comment]
I did the way you said it @pan-theo and my lowest is 15 brightness and highest is 0 brightness, can anybody confirm if that means vitabright works or doesn't work?

The same values are with vitabright in ux0 and ur0, so that doesn't seem to solve the problem, but definitely was worth trying out.

**Comment by pan-theo** [comment]
@olokos I don't know man, I wrote exactly what I did that worked for me. 
Meanwhile, this issue opened is only for some 1000 (1st model) Vitas. 

It's important to note that the directories used majorly depend on the type of storage used (could be official Vita card or SD2Vita). 
I use an official Sony Vita storage card, so all these instructions are based on that.

If you're using SD2Vita, then the formatting should be different, so maybe try putting the plugin in whatever config file works for you. I'm trying to at least point out the error in the instructions of the plugin's page, and also say that a different directory for plugins made it work.

If it's for 1000 model, official Sony card, double check with my instructions, are you sure you did it right? 

• config file in ux0:tai 
• in a line under *KERNEL 
• write ux0:tai/vitabright.skprx 
• make sure vitabright.skprx is placed in the same folder 
• re-boot 

There can always be factors I cannot see. But when I was re-checking and testing, these exact steps made it work, that's why I'm sharing it here. Sorry if it doesn't work for you. 

▪︎ Remember that the storage type affects the directories in your Vita, such as what is used for plugins.

**Comment by pan-theo** [comment]
Also, @olokos , what will show whether or not VitaBright works will be the noticeable new brightness levels. When it worked for me, there was 3 new levels before the (default) lowest one. The first will be a bit dimmer than the original (which is what is mostly needed), the 2nd one will be unexpectedly the dimmest, with a kinda weird colour (having extra green), and the 3rd (last) one, will be kinda like the first but with warmer colours, a bit like with a blue light filter. 

The 2nd can be fixed by appropriately adjusting some of the first values shown in the VitaBright editor. 


So, if you don't see any of these (new) things, then VitaBright has not worked yet.

**Comment by aorestr** [comment]
Hi there! Another proud user of a 100X model with the same issue as everyone! I tried the solution proposed by @pan-theo but it sadly didn't work. Isn't there any other workaround?

**Comment by ZioPao** [comment]
I've got a PCH-1004. In the log I've got the same supplier-id as @falkenhawk .  
Is this still "open" or is it a wontfix at this point? Just asking, not making any requests.

**Comment by MadZiontist** [comment]
I thought I had got it working on my PCH-1101 Vita on 3.60 Enso, but that was just a placebo effect. It doesn't work. I've disabled the 3g module. I've tried putting at the top of my tai config, and uninstalling all other plugins, but no dice. If you can post a link to the latest debug version, I can post the log file.

**Comment by sam22200** [comment]
Same here: doesn't work on my Vita 100X 3G
And I lived in Paris @devnoname120 !

**Comment by SKGleba** [comment]
precompiled binary (with fix from PR#28): 
[vitabright.zip](https://github.com/devnoname120/vitabright/files/6366269/vitabright.zip)

**Comment by SKGleba** [comment]
alternatively, instead of injecting, you can just write the table ptr to (SceOled+0x302c).
It seems like a cleaner solution especially if you free & inject repeatedly with the lut editor.

**Comment by ZioPao** [comment]
I can confirm @SKGleba binary works on my PCH-1004. The lut editor seems to crash though. Don't know if it's only me, but it works with the currently posted relase from devwithnoname.

**Comment by sam22200** [comment]
Wow thank you @SKGleba  
I confirm your version works on my PCH-1004 3G.

**Comment by MadZiontist** [comment]
> 
> 
> precompiled binary (with fix from PR#28):
> [vitabright.zip](https://github.com/devnoname120/vitabright/files/6366269/vitabright.zip

So far, so good...it's working on my PCH-1101. Previously the only version that worked was v0.01. This newer, fixed version works a lot better. Thanks @SKGleba!

**Comment by evillurker** [comment]
I confirm this works with 3 of my vitas - 1000, and 2000 @SKGleba thank you so much

**Comment by AciliBorek** [comment]
#29 Works for my 1104 while newest versions didnt. @SKGleba Thank you so much you really did a great thing.

**Comment by SKGleba** [comment]
Until it gets merged ive added a fixed skprx + lut editor in https://github.com/skgleba/vitabright/releases/latest

**Comment by evillurker** [comment]
> 
> 
> Until it gets merged ive added a fixed skprx + lut editor in https://github.com/skgleba/vitabright/releases/latest

You are the king!!

**Comment by Quad-Plex** [comment]
> precompiled binary (with fix from PR#28):
> [vitabright.zip](https://github.com/devnoname120/vitabright/files/6366269/vitabright.zip)

confirmed working on my PCH-1004. Big thanks!!

**Comment by JhonnySalles** [comment]
Its work in PCH-1010

**Comment by tonyzehs** [comment]
> Until it gets merged ive added a fixed skprx + lut editor in https://github.com/skgleba/vitabright/releases/latest

@SKGleba Thank you, you beautiful angel. Didn't work until I installed your modded version on my PCH1001/ZA10 Vita.

**Comment by devnoname120** [comment]
Can you guys try this new beta version of Vitabright?
[vitabright-2.0b2.zip](https://github.com/devnoname120/vitabright/files/8263575/vitabright-2.0b2.zip)

**Comment by SeongGino** [comment]
Unfortunately, neither the beta nor the recently released build works on my system (but still works with @SKGleba's mod).

**Comment by devnoname120** [comment]
@SeongGino Transfer the gamma table again via FTP. It probably fucked up in that step.

Excerpt from the README:

> **Important note**: If you want to use FileZilla for transferring the file vitabright_lut.txt, you'll first need to [change the transfer type from **Auto** to **Binary**](https://stackoverflow.com/a/555003) or it can prevent vitabright from parsing it correctly.

**Comment by SeongGino** [comment]
Of course, this had to be the perfect storm to cause me issue. ~~Last time I use FZ on Linux...~~
=w='
Fixed now, confirmed works, new lut table is perfect. So sorry for stirring an empty pot.

**Comment by devnoname120** [comment]
@SeongGino Np! I'll try to make the LUT parsing more resilient so that it can be properly read even with bad transfers.
